sendmail.py

```diff
@@ -1,56 +1,73 @@
-import email, smtplib, ssl
+import smtplib
+import ssl
 from email.mime.multipart import MIMEMultipart
 from email.mime.text import MIMEText
 
-class SendMail():
-  def __init__(self,credentials):
-    self.credentials = credentials
-    self.subject = self.credentials["subject"]
-    self.body = self.credentials["body"]
-    self.sender_email = self.credentials["sender_email"]
-    self.receiver_email = self.credentials["receiver_email"]
-    self.password = self.credentials["gmailpass"]
-    self.message = None
-
-  def setCredentials(self,statsDict):
-    # Create a multipart message and set headers
-    self.message = MIMEMultipart("alternative")
-    self.message["Subject"] = self.subject
-    self.message["From"] = self.sender_email
-    self.message["To"] = self.receiver_email
-
-    # Create the plain-text and HTML version of your message
-    text = f"""\
+
+class SendMail:
+    def __init__(self, credentials: dict):
+        self.credentials = credentials
+        self.subject = self.credentials["subject"]
+        self.body = self.credentials["body"]
+        self.sender_email = self.credentials["sender_email"]
+        self.receiver_email = self.credentials["receiver_email"]
+        self.password = self.credentials["gmailpass"]
+        self.message = None
+
+    def set_credentials(self, stats_dict: list):
+        # Create a multipart message and set headers
+        self.message = MIMEMultipart("alternative")
+        self.message["Subject"] = self.subject
+        self.message["From"] = self.sender_email
+        self.message["To"] = self.receiver_email
+
+        # Create the plain-text and HTML version of your message
+        text = f"""
     {self.body}
     
     OverAll Stats Is Shown Below :
 
-    Max Discount Found = {statsDict[0]["Max_Discount"]["Discount"]}, App Amount = {statsDict[0]["Max_Discount"]["Count"]}
-    Lowest App Price = {statsDict[0]["Lowest_Price"]["Price"]}, App Amount = {statsDict[0]["Lowest_Price"]["Count"]}
-    Highest App Price = {statsDict[0]["Highest_Price"]["Price"]}, App Amount = {statsDict[0]["Highest_Price"]["Count"]}
-    Free Apps Amount = {statsDict[0]["Free_Games"]["Count"]}
-    Free Apps = {"** No Free App Avaialble **" if statsDict[0]["Free_Games"]["Count"]==0 else str(statsDict[0]["Free_Games"]["Game_Names"])}
+    Max Discount Found = {stats_dict[0]["Max_Discount"]["Discount"]}, App Amount = {
+        stats_dict[0]["Max_Discount"]["Count"]}
+    Lowest App Price = {stats_dict[0]["Lowest_Price"]["Price"]}, App Amount = {
```

Comment:
So as you can see, the autoformatting sometimes can make things a little weird. You could shorten these lines by breaking out stat_dict[0] into it's on variable

Reply:
Okh. i will do that in future. I used those data only once. Thats why i didn't use any varibale. Thank you for your feedback